File: 165-compare-version-numbers/compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        vector<int> v1,v2;
        stringstream ss1(version1), ss2(version2);
        string s;
        while(getline(ss1,s,'.')){
            v1.push_back(stoi(s));
        }
        while(getline(ss2,s,'.')){
            v2.push_back(stoi(s));
        }
        int idx = 0, l1 =  v1.size(), l2 = v2.size();
        while(idx < l1 && idx < l2){
            if( v1[idx] > v2[idx]) return 1;
            else if (v1[idx] < v2[idx]) return -1;
            idx++;
        }
        while(idx < l1){
            if( v1[idx] > 0) return 1;
            idx++;
        }
        while(idx < l2){
            if (v2[idx] > 0) return -1;
            idx++;
        }
        return 0;
    }
};

File: 165-compare-version-numbers/compare-version-numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = [int(i) for i in version1.split(".")]
        v2 = [int(i) for i in version2.split(".")]
        l1,l2,i = len(v1), len(v2), 0
        while i < l1 and i < l2:
            if v1[i] > v2[i] : return 1
            elif v1[i] < v2[i] : return -1
            else : i += 1
        while i < l1:
            if v1[i] > 0: return 1
            else: i += 1
        while i < l2:
            if v2[i] > 0: return -1
            else: i += 1
        return 0